### DISCUSSION TITLE
mixcr buildLibrary process

### DISCUSSION THREAD

**Discussion by silvia1234567890** [milaboratory/mixcr#1576] [Q&A] [answered]
Hello,
Could you explain to me how MiXCR build the library for species that are not human, mouse,... And why MiXCR needs the NCBI taxon-id? I'm just trying to understand the process...
Would there be any way to build the library only with my own fasta files?
Thank you in advance!

**Comment by mizraelson** [comment, marked as answer]
Hi, so MiXCR tries to align the provided FASTA sequences to the build in known references to put the correct reference point for each FR and CDR in your FASTA. Taxon id is only required due to the format of the reference, MiXCR does not upload any more sequences from NCBI and only uses your FASTA files. In fact you can use any random 4 digit number as a taxon id and it will work.
